platform/linux-generic/odp_pool.c

```diff
@@ -622,9 +623,9 @@ static odp_pool_t pool_create(const char *name, const odp_pool_param_t *params,
 		pool->name[ODP_POOL_NAME_LEN - 1] = 0;
 	}
 
-	name_len = strlen(pool->name);
-	memcpy(uarea_name, pool->name, name_len);
-	strcpy(&uarea_name[name_len], postfix);
+	/* Format SHM names from prefix, pool index and pool name. */
+	sprintf(shm_name,   "pool_%03i_%s", pool->pool_idx, pool->name);
+	sprintf(uarea_name, "pool_%03i_uarea_%s", pool->pool_idx, pool->name);
```

Comment:
One issue with this change is that it breaks IPC pktio if pools aren't created in the same order in both processes.